config/initializers/devise.rb

```diff
@@ -276,14 +276,37 @@
   #   manager.default_strategies(scope: :user).unshift :some_external_strategy
   # end
 
+
+
+  Devise.add_module :remote_authenticatable, :controller => :sessions, :route => { :session => :routes }
+
   if Rails.env == 'production' || 
```

Comment:
I think this could read better if you extract some of the flow control into a method `authentication_strategy` that does some like:
```
if Rails.env == 'production' || ...
  {name: :remote, strategy: Devise::Strategies::RemoteAuthenticatable}
else
  {name: :test, strategy: Devise::Strategies::TestAuthenticatable}
end
```

and you could use it like:
```
config.warden do |manager|
  strategy = authentication_strategy
  manager.strategies.add(strategy[:name], strategy[:strategy])
  manager.default_strategies(:scope => :user).unshift strategy[:name]
end 
```

Reply:
if you provide an aws access key and secret key it will load up RemoteAuthenticatable regardless of environment.